### PR TITLE
PMM-11962 filtering for services and nodes in inventory

### PR DESCRIPTION
**Pull request by filipmikes1**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
https://jira.percona.com/browse/PMM-11962

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: public/app/percona/inventory/Tabs/Agents.tsx [hunk 1]
```diff
@@ -218,22 +249,22 @@ export const Agents: FC<GrafanaRouteComponentProps<{ id: string }>> = ({ match }
...
+            totalItems={flattenAgents.length}
             rowSelection
             autoResetSelectedRows={false}
             onRowSelection={handleSelectionChange}
             showPagination
             pageSize={25}
             allRowsSelectionMode="page"
             emptyMessage={Messages.agents.emptyTable}
-            emptyMessageClassName={styles.emptyMessage}
+            // emptyMessageClassName={styles.emptyMessage}
```
Comment (fabio-silva) [inline public/app/percona/inventory/Tabs/Agents.tsx:262]: if this not needed?

File: public/app/percona/inventory/Tabs/Nodes.tsx [hunk 2]
```diff
@@ -37,23 +38,29 @@ export const NodesTab = () => {
...
       {
         Header: Messages.nodes.columns.nodeId,
-        accessor: (row) => row.params.nodeId,
+        accessor: 'nodeId',
+        type: FilterFieldTypes.TEXT,
       },
       {
         Header: Messages.nodes.columns.nodeType,
         accessor: 'type',
+        type: FilterFieldTypes.TEXT,
```
Comment (fabio-silva) [inline public/app/percona/inventory/Tabs/Nodes.tsx:58]: should this be text? I think the believe the type is an enum

File: public/app/percona/inventory/Tabs/Services.tsx [hunk 3]
```diff
@@ -67,59 +83,93 @@ export const Services = () => {
...
   const columns = useMemo(
-    (): Array<Column<Service>> => [
+    (): Array<ExtendedColumn<FlattenService>> => [
       {
         Header: Messages.services.columns.status,
-        accessor: (row) => row.params.status,
+        accessor: 'status',
         Cell: ({ value }: { value: ServiceStatus }) => (
           <Badge
             text={capitalizeText(value)}
```
Comment (fabio-silva) [inline public/app/percona/inventory/Tabs/Services.tsx:106]: Let's take the chance to fix something here:
```
text={value === ServiceStatus.NA ? ServiceStatus.NA : capitalizeText(value)}
```

Comment (fabio-silva) [reply public/app/percona/inventory/Tabs/Services.tsx:106]: Because it's showing as "N/a" right now

**Review by fabio-silva — CHANGES_REQUESTED** [review]

File: public/app/percona/integrated-alerting/components/Table/Filter/Filter.utils.ts [hunk 4]
```diff
@@ -114,7 +114,7 @@ export const isValueInTextColumn = (
 };
 
 export const isTextIncluded = (needle: string, haystack: string): boolean =>
-  haystack.toLowerCase().includes(needle.toLowerCase());
+  haystack.toString().toLowerCase().includes(needle.toLowerCase());
```
[the remark this answers is not in the document]
Comment (filipmikes1) [reply public/app/percona/integrated-alerting/components/Table/Filter/Filter.utils.ts:117]: Because in runtime when value is number it will fail

**Review by fabio-silva — DISMISSED** [review]

**Review by fabio-silva — APPROVED** [review]

**Pull request merged**